fontbe/src/kern.rs

```diff
@@ -140,6 +147,96 @@ impl Work<Context, AnyWorkId, Error> for GatherIrKerningWork {
     }
 }
 
+/// 'align' the kerning, ensuring each pair is defined for each location.
+///
+/// missing pairs are filled in via the UFO kerning value lookup algorithm:
+///
+/// <https://unifiedfontobject.org/versions/ufo3/kerning.plist/#kerning-value-lookup-algorithm>
+fn align_kerning(
+    groups: &KerningGroups,
+    instances: &mut HashMap<NormalizedLocation, KerningInstance>,
+) {
+    let union_kerning = instances
+        .values()
+        .flat_map(|instance| instance.kerns.keys())
+        .cloned()
+        .collect::<HashSet<_>>();
+
+    let side1_glyph_to_group_map = groups
+        .groups
+        .iter()
+        .filter(|(group, _)| matches!(group, KernGroup::Side1(_)))
+        .flat_map(|(group, glyphs)| glyphs.iter().map(move |glyph| (glyph, group)))
+        .collect::<HashMap<_, _>>();
+    let side2_glyph_to_group_map = groups
+        .groups
+        .iter()
+        .filter(|(group, _)| matches!(group, KernGroup::Side2(_)))
+        .flat_map(|(group, glyphs)| glyphs.iter().map(move |glyph| (glyph, group)))
+        .collect::<HashMap<_, _>>();
+
+    for instance in instances.values_mut() {
+        let missing_pairs = union_kerning
+            .iter()
+            .filter(|pair| !instance.kerns.contains_key(pair))
+            .collect::<Vec<_>>();
+
+        for pair in missing_pairs {
+            let value = lookup_kerning_value(
+                pair,
+                instance,
+                &side1_glyph_to_group_map,
+                &side2_glyph_to_group_map,
+            );
+            log::info!("got aligned value {value} for kern pair {pair:?}");
```

Comment:
maybe log::info level would be too noisy for this, debug might work better

Reply:
oh good call, I just added this for my personal edification so I could see if what was happening looked sensible, I think we can just remove it.